core/lsp/helpers.cc

```diff
@@ -123,6 +126,7 @@ string prettyDefForMethod(const core::GlobalState &gs, core::MethodRef method) {
         methodNamePrefix = "self.";
     }
     vector<string> prettyArgs;
+    auto defaultArgumentPlaceholder = concretizeIfAbstract ? "T.let(T.unsafe(nil), T.untyped)" : "…";
```

Comment:
You're using `T.let(T.unsafe(nil), T.untyped)`, which is redundant—`T.unsafe(nil)` is already `T.untyped`. On top of that, I don't even think we want to use `T.unsafe(nil)`. Sorbet autocorrects never insert `T.unsafe` unless the `--suggest-unsafe` command line option has been passed. See [Silencing errors in bulk](https://sorbet.org/docs/cli#silencing-errors-in-bulk)—the idea is that we only expose `T.unsafe` in autocorrects to those who are doing bulk codemods, where they don't have the time to correctly fix individual errors, and need something that will Just Work.

Some alternatives:

- Check if `NilClass` is a subtype of the expected type, and generate `"nil"` directly.

  If it's not, fall back to one of the options below.
  
- Translate it to `_` instead.

  Syntactically, this would represent a call to a method named, the same as if the user had typed `self._()`. This method is unlikely to exist, and so in `# typed: true` files or higher Sorbet will show a "method does not exist" error.
  
- Omit the default.

  This will generate a new type error, that the user will see after accepting the autocorrect, alerting them that they need to take an explicit action to resolve this. Unlike the previous, this will get reported even in `# typed: false` files.
  
  [example on sorbet.run](https://sorbet.run/#%23%20typed%3A%20false%0A%0Aclass%20Parent%0A%20%20extend%20T%3A%3ASig%0A%20%20extend%20T%3A%3AHelpers%0A%0A%20%20abstract!%0A%0A%20%20sig%20%7Babstract.params%28x%3A%20Integer%29.void%7D%0A%20%20def%20example%28x%3A%200%29%0A%20%20end%0Aend%0A%0Aclass%20Child%20%3C%20Parent%0A%0A%20%20sig%20%7Boverride.params%28x%3A%20Integer%29.void%7D%0A%20%20def%20example%28x%3A%29%0A%20%20end%0A%0Aend)
  
- Start with the "omit the default" option above, and combine it with using `T.unsafe(nil)` as the default only if `--suggest-unsafe` was passed.

Reply:
> Start with the "omit the default" option above, and combine it with using T.unsafe(nil) as the default only if --suggest-unsafe was passed.

I implemented this alternative in 5488e0e.